src/contrib/mlir/utils.cpp

```diff
@@ -35,15 +46,47 @@ static llvm::cl::opt<bool> clPrintIRAfterAll(
 
 void ngraph::runtime::ngmlir::initializeNGraphMLIR()
 {
-    // Initialize a dialect only once.
-    // We currently have no way to query if a dialect is previously
-    // registered. So using a global flag instead.
-    static bool init = false;
-    if (!init)
-    {
-        mlir::registerDialect<mlir::NGraphOpsDialect>();
-        init = true;
-    }
+    // Initialize MLIR dialects and passes only once.
+    static bool init_once = []() {
+        // In-tree Dialects.
+        registerDialect<AffineOpsDialect>();
+        registerDialect<LLVM::LLVMDialect>();
+        registerDialect<loop::LoopOpsDialect>();
+        registerDialect<StandardOpsDialect>();
+        registerDialect<vector::VectorOpsDialect>();
+        
+        // nGraph dialects. 
+        registerDialect<mlir::NGraphOpsDialect>();
+
+        // In-tree passes.
+        // NO-OPT to avoid DCE on the following pass initializations.
+        if (std::getenv("bar") != (char*)-1)
```

Comment:
what is this for ?

Reply:
Copy-paste from `mlir-opt` to avoid DCE of the code below. Fixing typo: NO-OPT -> No-op